socket-ballerina/build.gradle

```diff
@@ -0,0 +1,130 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import org.apache.tools.ant.taskdefs.condition.Os
+
+apply from: "$rootDir/gradle/mvnPublisher.gradle"
+
+description = 'Ballerina - Socket Ballerina Generator'
+
+configurations {
+    balTools
+    socketBallerinaBirDir
+    socketBallerinaJar
+}
+
+dependencies {
+    balTools ("org.ballerinalang:jballerina-tools:${ballerinaLangVersion}") {
+        transitive = false
+    }
+    compile project(':socket-native')
+}
+
+
+
+clean {
+    delete "$project.projectDir/target"
+    delete "$project.projectDir/Ballerina.lock"
+}
+
+jar {
+    manifest {
+        attributes('Implementation-Title': project.name,
+                'Implementation-Version': project.version)
+    }
+}
+
+task unzip {
+    doLast {
+        configurations.balTools.resolvedConfiguration.resolvedArtifacts.each { artifact ->
+            copy {
+                from project.zipTree(artifact.getFile())
+                into new File(project.buildDir, "extracted-distribution/")
+            }
+        }
+    }
+}
+
+def artifactSocketBallerinaBirDir = file("$project.projectDir/target/caches/bir_cache/")
+def artifactSocketBallerinaJar = file("$project.projectDir/target/caches/jar_cache/ballerina/socket/ballerina.socket.jar")
+def balSockerJar =  file("$project.projectDir/build/libs/ballerina-socket-${ballerinaLangVersion}.jar")
+
+task ballerinaBuild {
+    dependsOn unzip
+    inputs.dir file(project.projectDir)
+    doLast {
+        exec {
+            workingDir project.projectDir
+            environment "JAVA_OPTS", "-DBALLERINA_DEV_COMPILE_BALLERINA_ORG=true"
+            def distributionBinPath =  project.projectDir.absolutePath + "/build/extracted-distribution/jballerina-tools-${ballerinaLangVersion}/bin"
+            if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+                commandLine 'cmd', '/c', "$distributionBinPath/ballerina.bat build -c -a"
+            } else {
+                commandLine 'sh', '-c', "$distributionBinPath/ballerina build -c -a"
+            }
+        }
+        copy {
+            // Copying the jar from ballerina build
+            from "$project.projectDir/target/caches/jar_cache/ballerina/socket/ballerina-socket-.jar"
+            into "$project.projectDir/target/caches/jar_cache/ballerina/socket/"
+            rename { String fileName ->
+                fileName.replace("ballerina-socket-.jar", artifactSocketBallerinaJar.getName())
+            }
+        }
+        // Copy the BIR to the 0.0.0 folder within it
+        ant.move file: "${artifactSocketBallerinaBirDir}/ballerina/socket/socket.bir",
+                todir: "${artifactSocketBallerinaBirDir}/ballerina/socket/0.0.0/"
```

Comment:
What is this 0.0.0? Is that something internal or socket module version?

Reply:
This is relevant to the `bir-cache` directory structure 
```
jballerina-1.3.1-SNAPSHOT/bir-cache/ballerina/socket/0.0.0/socket.bir
```